### ISSUE TITLE
Elasticsearch hadpop error :java.lang.NoClassDefFoundError: Could not initialize class org.elasticsearch.spark.serialization.ScalaMetadataExtractor

### ISSUE THREAD

**Issue by vishnugs** [elastic/elasticsearch-hadoop#1558]
### Issue description

Description
Unable to use elasticsearch-hadoop jar with spark 3.0.0 as it needs scala 2.12.
 
### Steps to reproduce

Code:

```
  <dependency>
            <groupId>org.elasticsearch</groupId>
            <artifactId>elasticsearch-hadoop</artifactId>
            <version>${elastic.version}</version>
            <scope>compile</scope>
        </dependency>
```

Strack trace:
Receiving following error after upgrading to spark 3.0.0 with any of the version of elasticsearch-hadoop jar

tried elastic.version with 7.1.1 , 7.4.0,7.5.0 , 7.10.0

```
2020-11-20 14:54:25 INFO  DAGScheduler:57 - ResultStage 89 (runJob at EsSpark.scala:108) failed in 0.102 s due to Job aborted due to stage failure: Task 1 in stage 89.0 failed 4 times, most recent failure: Lost task 1.3 in stage 89.0 (TID 517, 10.4.6.202, executor 2): java.lang.NoClassDefFoundError: Could not initialize class org.elasticsearch.spark.serialization.ScalaMetadataExtractor$
        at org.elasticsearch.spark.serialization.ScalaMetadataExtractor.getValue(ScalaMetadataExtractor.scala:32)
        at org.elasticsearch.hadoop.serialization.bulk.PerEntityPoolingMetadataExtractor.get(PerEntityPoolingMetadataExtractor.java:97)
        at org.elasticsearch.hadoop.serialization.bulk.AbstractBulkFactory.getMetadataExtractorOrFallback(AbstractBulkFactory.java:493)
        at org.elasticsearch.hadoop.serialization.bulk.AbstractBulkFactory.writeObjectHeader(AbstractBulkFactory.java:404)
        at org.elasticsearch.hadoop.serialization.bulk.AbstractBulkFactory$DynamicHeaderRef.getDynamicContent(AbstractBulkFactory.java:177)
        at org.elasticsearch.hadoop.serialization.bulk.TemplatedBulk.writeTemplate(TemplatedBulk.java:84)
        at org.elasticsearch.hadoop.serialization.bulk.TemplatedBulk.write(TemplatedBulk.java:56)
        at org.elasticsearch.hadoop.serialization.bulk.BulkEntryWriter.writeBulkEntry(BulkEntryWriter.java:68)
        at org.elasticsearch.hadoop.rest.RestRepository.writeToIndex(RestRepository.java:170)
        at org.elasticsearch.spark.rdd.EsRDDWriter.write(EsRDDWriter.scala:74)
        at org.elasticsearch.spark.rdd.EsSpark$$anonfun$doSaveToEs$1.apply(EsSpark.scala:108)
        at org.elasticsearch.spark.rdd.EsSpark$$anonfun$doSaveToEs$1.apply(EsSpark.scala:108)
        at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
        at org.apache.spark.scheduler.Task.run(Task.scala:127)
        at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:444)
        at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377)
        at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:447)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

``` 

### Version Info

OS:         :   Ubuntu
JVM         :   Java 8
Hadoop/Spark:  Spark 3.0.0
ES-Hadoop   :  7.1.1 through 7.5.1 
ES          :  7.5

Is it due to conflct scala version ?How this can be corrected with scala ?

**Comment by jsteggink** [comment]
I ran into the same issue. It's fixed in 7.11.0 as the 2.12 versions will be built. However, you'll have to wait until it's released or get a snapshot. I tried to get a snapshot, but the SonaType releases seem to be zip only instead of jar, no idea why.

**Comment by jbaiera** [comment]
Spark 3.0 support will be landing in v7.12, supporting Scala 2.12
